evmbin/src/display/json.rs

```diff
@@ -123,19 +103,19 @@ impl trace::VMTracer for Informant {
 	fn trace_executed(&mut self, gas_used: U256, stack_push: &[U256], mem_diff: Option<(usize, &[u8])>, store_diff: Option<(U256, U256)>) {
 		let info = ::evm::Instruction::from_u8(self.instruction).map(|i| i.info());
 
-		let trace = format!(
-			"{{\"pc\":{pc},\"op\":{op},\"opName\":\"{name}\",\"gas\":\"0x{gas:x}\",\"gasCost\":\"0x{gas_cost:x}\",\"memory\":{memory},\"stack\":{stack},\"storage\":{storage},\"depth\":{depth}}}",
-			pc = self.pc,
-			op = self.instruction,
-			name = info.map(|i| i.name).unwrap_or(""),
-			gas = gas_used.saturating_add(self.gas_cost),
-			gas_cost = self.gas_cost,
-			memory = self.memory(),
-			stack = self.stack(),
-			storage = self.storage(),
-			depth = self.depth,
-		);
-		self.traces.push(trace);
+		let trace = json!({
+			"pc": self.pc,
+			"op": self.instruction,
+			"opName": info.map(|i| i.name).unwrap_or(""),
+			"gas": format!("{:#x}", gas_used.saturating_add(self.gas_cost)),
+			"gasCost": format!("{:#x}", self.gas_cost),
+			"memory": format!("0x{}", self.memory.to_hex()),
+			"stack": self.stack,
+			"storage": self.storage,
```

Comment:
I think this implementation is different than it used to be (`Display` vs `Debug`). It's a shame we don't have tests with non-empty `storage`.

Reply:
Why don't any of the tests have a non-empty `storage`?

Reply:
@ordian So both `stack` and `storage` don't implement `Display` (since I removed the `stack()` and `storage()` functions). Why is having `Display` important for these variables?

Reply:
Sorry for not being clear. What I mean was, that the `fn storage()` was using `Debug` (`0x{:?}`) key/value representation, and I thought that serde impl was using `Display` one, but I was wrong.
Thanks for adding an additional test!

Reply:
Ah, I see. No problem!